src/main/java/com/example/auctionapp/customValidator/BidVerifierValidator.java

```diff
@@ -0,0 +1,64 @@
+package com.example.auctionapp.customValidator;
+
+import com.example.auctionapp.Util.TimeUtility;
+import com.example.auctionapp.exception.BadRequestException;
+import com.example.auctionapp.model.Bid;
+import com.example.auctionapp.model.Product;
+import com.example.auctionapp.repository.ProductRepository;
+import org.springframework.beans.BeanWrapperImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.List;
+
+public class BidVerifierValidator implements ConstraintValidator<BidVerifier, Object> {
+
+    String productId;
+    String bidTime;
+    String bidAmount;
+    private final ProductRepository productRepository;
+
+    @Autowired
+    public BidVerifierValidator(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    public void initialize(BidVerifier constraintAnnotation) {
+        this.productId = constraintAnnotation.productId();
+        this.bidTime = constraintAnnotation.bidTime();
+        this.bidAmount = constraintAnnotation.bidAmount();
+    }
+
+    public boolean isValid(Object value,
+                           ConstraintValidatorContext context) {
+
+        try {
+            Long productIdValue = Long.parseLong(new BeanWrapperImpl(value)
+                    .getPropertyValue(productId).toString());
+            Long bidTimeValue = Long.parseLong(new BeanWrapperImpl(value)
+                    .getPropertyValue(bidTime).toString());
+            Double bidAmountValue = Double.parseDouble(new BeanWrapperImpl(value)
+                    .getPropertyValue(bidAmount).toString());
+
+            Product product = productRepository.findById(productIdValue);
```

Comment:
Validator annotations are used to validate in the scope of the given object - they validate on the properties of that object etc, sometimes they accept some enum values to validate or lists...
That is why they are configured as a simple annotation.
If the validation needs some kind of complex pre-configuring (like initialising a repository) it should not be placed in this kind of validation. Put this in the service where you added it initially.